src/javascript/app_2/Stores/Modules/Trading/Helpers/proposal.js

```diff
@@ -40,10 +40,16 @@ const createProposalRequestForContract = (store, type_of_contract) => {
     const obj_expiry = {};
     if (store.expiry_type === 'endtime') {
         const expiry_date = moment.utc(store.expiry_date);
-        const start_date  = moment.unix(store.start_date || (store.root_store.common.server_time / 1000)).utc();
-        const is_same_day = expiry_date.isSame(start_date, 'day');
-        const expiry_time = is_same_day ? store.expiry_time : '23:59:59';
-        obj_expiry.date_expiry = convertToUnix(expiry_date.unix(), expiry_time);
+        const start_time = store.start_time.split(':');
+        if (store.contract_start_type === 'spot') {
+            const expiry_time = store.expiry_time;
+            obj_expiry.date_expiry = convertToUnix(expiry_date.unix(), expiry_time);
+        } else {
+            const start_date  = moment.unix(store.start_date || (store.root_store.common.server_time / 1000)).utc();
+            const is_same_day = expiry_date.isSame(start_date, 'day');
```

Comment:
Since `is_same_day` is not relevant to the `if` statement, we can promote it to upper scope.